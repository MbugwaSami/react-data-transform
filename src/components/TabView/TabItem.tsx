import { FC } from "react";

import "./tabView.scss";

const TabItem: FC<TabItemProps> = (props) => {
  const { title, onClick, active, styles, icon } = props;

  return (
    <div
      style={styles}
      aria-label={title}
      tabIndex={0}
      role="presentation"
      className={active ? "rdt-tab-item-selected rdt-tab-item" : "rdt-tab-item"}
      onClick={onClick}
      onKeyPress={(e) => {
        if (e.key === "Enter" || e.key === " ") {
          onClick();
        }
      }}
    >
      <div className="rdt-tab-item-icon">{icon && icon}</div>
      <h4>{title}</h4>
    </div>
  );
};

export default TabItem;
