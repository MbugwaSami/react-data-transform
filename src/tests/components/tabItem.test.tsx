import React from "react";
import { shallow, render, mount } from "enzyme";
import TabItem from "../../components/TabView/TabItem";

describe("<TabItem />", () => {
  const onTabClickMock = jest.fn();
  const renderTabIcon = () => {
    return <div></div>;
  };
  const props = {
    title: "title",
    active: false,
    icon: renderTabIcon(),
    onClick: onTabClickMock,
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should render <TabItem />", () => {
    const wrapper = shallow(<TabItem {...props} />);
    const tabElement = wrapper.find(".rdt-tab-item");
    expect(tabElement).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
  it("should call onClick", () => {
    const wrapper = shallow(<TabItem {...props} />);
    const tabElement = wrapper.find(".rdt-tab-item");
    tabElement.simulate("click");
    expect(onTabClickMock).toHaveBeenCalledTimes(1);
  });
  it("should call onClick when enter is pressed", () => {
    const wrapper = shallow(<TabItem {...props} />);
    const tabElement = wrapper.find(".rdt-tab-item");
    tabElement.simulate("keypress", { key: "Enter" });
    expect(onTabClickMock).toHaveBeenCalledTimes(1);
  });
  it("should call onClick when space is pressed", () => {
    const wrapper = shallow(<TabItem {...props} />);
    const tabElement = wrapper.find(".rdt-tab-item");
    tabElement.simulate("keypress", { key: " " });
    expect(onTabClickMock).toHaveBeenCalledTimes(1);
  });
  it("should add rdt-tab-item-selected class if active ", () => {
    const wrapper = shallow(<TabItem {...props} />);
    wrapper.setProps({ active: true });
    const activeElement = wrapper.find(".rdt-tab-item-selected");
    expect(activeElement).toHaveLength(1);
  });
});
