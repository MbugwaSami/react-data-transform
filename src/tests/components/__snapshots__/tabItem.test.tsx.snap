// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabItem /> should render <TabItem /> 1`] = `
<div
  aria-label="title"
  className="rdt-tab-item"
  onClick={[MockFunction]}
  onKeyPress={[Function]}
  role="presentation"
  tabIndex={0}
>
  <div
    className="rdt-tab-item-icon"
  >
    <div />
  </div>
  <h4>
    title
  </h4>
</div>
`;
